packages/git/src/browser/git-quick-open-service.ts

```diff
@@ -175,7 +179,7 @@ export class GitQuickOpenService {
                     const branches = await this.getBranches();
                     const executeBranch = async (branchItem: GitQuickOpenItem<Branch>) => {
                         try {
-                            await this.git.pull(repository, { remote: remoteItem.ref, branch: branchItem.ref.nameWithoutRemote });
+                            await this.git.pull(repository, { remote: remoteItem.ref.name, branch: branchItem.ref.nameWithoutRemote });
```

Comment:
Not sure how this happens, but this breaks the `pull` operation.

Please set a breakpoint [here](https://github.com/theia-ide/theia/blob/7ca74cd789e629f9477b8755d201b1312b17e387/packages/git/src/node/dugite-git.ts#L517) and look at the value of `currentRemote`. It supposed to be a string, but I see:
```
 {
  "name": "origin",
  "fetch": "https://github.com/theia-ide/theia.git",
  "push": "https://github.com/theia-ide/theia.git"
 }
```

Reply:
@AlexTugarev 
I wasn't able to reproduce the issue :disappointed: 
I've tried both with a repository with a single remote, as well as multiple remotes but I still can't reproduce the issue about having the `Remote` object at this point instead of the string.

Reply:
@vince-fugnitto, can you try it in gitpod?

I did the followings:
 - Open the PR in Gitpod: https://gitpod.io/#https://github.com/theia-ide/theia/pull/4190
 - Wait for the build, then execute `yarn --cwd ./examples/browser/ start --hostname 0.0.0.0 --port 3000` in the terminal.
 - Click on `Open Browser`.
 - Open `theia` in the workspace.
 - Go to the Git view and `pull`.

If you have any questions, let me know. Thank you!

Reply:
Thanks @kittaakos!

Reply:
thanks @AlexTugarev @kittaakos I'll take a look and see what the issue is

Reply:
I tried again and seem to get the correct value for `currentRemote`, in this case "origin".

![image](https://user-images.githubusercontent.com/40359487/52425576-4cb1ec80-2aca-11e9-98b9-e04338cd586d.png)

Is there something I may be doing incorrectly?

Reply:
> Is there something I may be doing incorrectly?

No idea.

I had a `Request pull failed with message: Authentication failed. You may not have permission to access the repository or the repository may have been archived.` error. I have to look into it again... Bear with me!

Reply:
@kittaakos did you get the chance to look into the issue again?

Reply:
I am checking it once more...